### PR TITLE
Fix some uninitialized variable warnings.

### PR DESCRIPTION
**Pull request by drwells**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass? (I didn't run IBFE tests since this is completely isolated)
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by drwells** [comment]
Fixed!

File: src/wave_generation/IrregularWaveBcCoef.cpp [hunk 1]
```diff
@@ -196,13 +196,14 @@ IrregularWaveBcCoef::setBcCoefs(Pointer<ArrayData<NDIM, double> >& acoef_data,
...
 #endif
 
-        double dof_posn[NDIM];
         for (Box<NDIM>::Iterator b(bc_coef_box); b; b++)
         {
             const SAMRAI::hier::Index<NDIM>& i = b();
             if (acoef_data) (*acoef_data)(i, 0) = 1.0;
             if (bcoef_data) (*bcoef_data)(i, 0) = 0.0;
 
+            IBTK::Point dof_posn;
```
Comment (boyceg) [inline src/wave_generation/IrregularWaveBcCoef.cpp:205]: Does `Point` have a default constructor?

Comment (drwells) [reply src/wave_generation/IrregularWaveBcCoef.cpp:205]: Surely you mean a not-default constructor?

The online documentation for `Matrix(const Scalar &s)` states (probably incorrectly) that it creates a 1x1 matrix, so I went for the unambiguous option.

Comment (boyceg) [reply src/wave_generation/IrregularWaveBcCoef.cpp:205]: I think what I meant was --- a "reasonable default constructor".

Comment (drwells) [reply src/wave_generation/IrregularWaveBcCoef.cpp:205]: The default constructor doesn't initialize data, so its not going to help in this case. Besides - we should throw an error if we read a value which we didn't set, so `signaling_NaN()` is the right choice here.

**Pull request merged**